platform/location_service.cpp

```diff
@@ -71,6 +75,8 @@ class PositionFilter
     {
 #if defined(OMIM_OS_MAC)
       m_services.push_back(CreateAppleLocationService(*this));
+#elif defined(OMIM_OS_LINUX)
+      m_services.push_back(CreateQtLocationService(*this, "geoclue2"));
```

Comment:
> Are there other services except geoclue2? 

It's an extensible plugin system. So yes, there are many other plugins out there already besides  `geoclue2`, some from Qt directly, some from third parties and we could also write our own.

For Linux the options that are worth listing are:
- Qt shipped [NMEA](https://doc.qt.io/qt-6/position-plugin-nmea.html)
- Qt shipped [gypsy](https://doc.qt.io/qt-6/position-plugin-gypsy.html)
- 3PP [gpsd](https://github.com/jmechnich/qtposition_gpsd)

> Why is it made as a parameter?

Mainly so that if we want to support any from the above list, then we can do so easily.
And theoretically for MacOS we could refactor and simplify the existing positioning to use this new `qt_location_service` with [corelocation](https://doc.qt.io/qt-6/position-plugin-corelocation.html).
And for Windows, there might also be something out there if anyone is even interested.

Reply:
Can several services be used at the same time for better location updates?

Reply:
Well yes. Perhaps one can imagine rare corner cases, in which it might make an actual difference. Let's say that `geoclue2` uses the wifi device to get a street level accuracy quickly which allows the user to hit the road already, until a `gpsd` only  high-end receiver boots up and starts providing the much more accurate positioning.